### PR TITLE
080624.01/comment and diff navigation

### PR DESCRIPTION
**Pull request by chidozieononiwu**
- Add feature for navigating comments and diffs on the review page.
CommentNavigation

DiffNavigation

### CODE REVIEW

**Comment by praveenkuttappan** [comment]
Let's discuss about this feature. I would prefer to see next and previous comment button like in the old system. Or even a button that becomes visible when you roll over the mouse. It does not feel like a right fit to put this in review page options panel.

**Comment by chidozieononiwu** [comment]
> Let's discuss about this feature. I would prefer to see next and previous comment button like in the old system. Or even a button that becomes visible when you roll over the mouse. It does not feel like a right fit to put this in review page options panel.

The advantage here is that the user does not need to find the first comment on the page before beginning to navigate. In some cases the first comment might be out of the view port. Also implementing a similar UI for diff is a bit cumbersome so I tough its best to put all the navigation UI in one place.

**Comment by praveenkuttappan** [comment]
> > Let's discuss about this feature. I would prefer to see next and previous comment button like in the old system. Or even a button that becomes visible when you roll over the mouse. It does not feel like a right fit to put this in review page options panel.
> 
> The advantage here is that the user does not need to find the first comment on the page before beginning to navigate. In some cases the first comment might be out of the view port. Also implementing a similar UI for diff is a bit cumbersome so I tough its best to put all the navigation UI in one place.

I see your point. One problem with this approach is that when a user decides to hide the right-side panel to get more real estate then they won't be able to use this feature. So, they will be forced to use the panel to jump to next diff or comment.  Can we a projected button at the top when scrolling over to go to first comment or diff similar to the icon button you had earlier added in revisions page.

File: src/dotnet/APIView/ClientSPA/src/app/_components/review-page-options/review-page-options.component.html [hunk 1]
```diff
@@ -67,6 +67,29 @@
...
 </app-page-options-section>
 
+<app-page-options-section sectionName="Find on Page">
+    <ul class="list-group">
+        <li class="list-group-item">                
+            <label class="small mx-1 fw-semibold">Comment:</label>
+            <div class="d-grid gap-2">
+                <div class="btn-group btn-group-sm" role="group" aria-label="Comment Navigation">
+                    <button type="button" class="btn btn-outline-secondary" (click)="commentThreadNavaigationEmitter.emit(CodeLineRowNavigationDirection.next)"><i class="bi bi-arrow-down"></i>Next</button>
+                    <button type="button" class="btn btn-outline-secondary" (click)="commentThreadNavaigationEmitter.emit(CodeLineRowNavigationDirection.prev)"><i class="bi bi-arrow-up"></i>Prev</button>
```
Comment (maririos) [inline src/dotnet/APIView/ClientSPA/src/app/_components/review-page-options/review-page-options.component.html:77]: can you add a tooltip so the complete message shows when hovering? i.e. `Previous`

File: src/dotnet/APIView/ClientSPA/src/app/_components/revisions-list/revisions-list.component.ts [hunk 2]
```diff
@@ -506,7 +507,7 @@ export class RevisionsListComponent implements OnInit, OnChanges {
           `Run <code>dotnet pack</code>`, 
           `Upload the resulting .nupkg file.`
         ];
-        this.acceptedFilesForReviewUpload = ".nupkg";
+        this.acceptedFilesForReviewUpload = ".nupkg, .dll";
```
Comment (maririos) [inline src/dotnet/APIView/ClientSPA/src/app/_components/revisions-list/revisions-list.component.ts:510]: so this is a current bug in prod for manual uploads?

Comment (chidozieononiwu) [reply src/dotnet/APIView/ClientSPA/src/app/_components/revisions-list/revisions-list.component.ts:510]: This is weird. let me do a rebase. Main is fine

**Comment by chidozieononiwu** [comment]
> I talked offline with @chidozieononiwu that we can leave the diff arrows only on the side panel. The comment ones can happen both in the comments and in the side panel. Could you update the images with that change?
> 
> Looking at the imagines, what do you think about swapping the order to show prev on the left and next in the right? I was looking at some UI designs and they mentioned to put next on the right and to give only one button preference. https://fluent2.microsoft.design/components/web/react/button/usage

Resolved

File: src/dotnet/APIView/ClientSPA/src/app/_components/revisions-list/revisions-list.component.ts (hunk 2 above)
Comment (maririos) [reply src/dotnet/APIView/ClientSPA/src/app/_components/revisions-list/revisions-list.component.ts:510]: weird. it is still showing up here

**Review by maririos — APPROVED** [review]

**Pull request merged**